pkg/frontend/v2/frontend_test.go

```diff
@@ -64,22 +66,35 @@ func setupFrontend(t testing.TB, reg prometheus.Registerer, schedulerReplyFunc f
 }
 
 func setupFrontendWithConcurrencyAndServerOptions(t testing.TB, reg prometheus.Registerer, schedulerReplyFunc func(f *Frontend, msg *schedulerpb.FrontendToScheduler) *schedulerpb.SchedulerToFrontend, concurrency int, logger log.Logger, opts ...grpc.ServerOption) (*Frontend, *mockScheduler) {
-	l, err := net.Listen("tcp", "localhost:0")
+	// We need to use different ports as the scheduler does not need the user header interceptor,
+	// whereas the frontend does.
+	frontendListener, err := net.Listen("tcp", "localhost:0")
 	require.NoError(t, err)
 
-	server := grpc.NewServer(opts...)
+	schedulerListener, err := net.Listen("tcp", "localhost:0")
+	require.NoError(t, err)
+
+	schedulerOpts := opts
+	frontendOpts := append(
+		opts,
```

Comment:
Won't this use the memory from `opts` if it has enough space for the additional interceptors? It's not clear to me if this will end up modifying `schedulerOpts`.